dataset.go

```diff
@@ -0,0 +1,389 @@
+package gdal
+
+/*
+#include "go_gdal.h"
+#include "gdal_version.h"
+
+#cgo linux  pkg-config: gdal
+#cgo darwin pkg-config: gdal
+#cgo windows LDFLAGS: -Lc:/gdal/release-1600-x64/lib -lgdal_i
+#cgo windows CFLAGS: -IC:/gdal/release-1600-x64/include
+*/
+import "C"
+import (
+	"fmt"
+	"unsafe"
+)
+
+// Get the driver to which this dataset relates
```

Comment:
It will be a long road, but probably worth making these function comments godoc style